inspector/src/components/actionTabs/tabs/propertyGrids/animations/addAnimation.tsx

```diff
@@ -26,17 +29,95 @@ export class AddAnimation extends React.Component<
     animationType: string;
     loopMode: number;
     animationTargetPath: string;
+    isUpdating: boolean;
   }
 > {
   constructor(props: IAddAnimationProps) {
     super(props);
-    this.state = {
-      animationName: '',
-      animationTargetPath: '',
-      animationType: 'Float',
-      loopMode: Animation.ANIMATIONLOOPMODE_CYCLE,
-      animationTargetProperty: '',
-    };
+
+    if (this.props.selectedToUpdate !== undefined) {
+      this.state = {

```

Comment:
This and the next block from line 64 look alike any way to factor ???

Reply:
Sure, I see what you mean.